include/cufinufft/impl.h

```diff
@@ -114,8 +114,11 @@ int cufinufft_makeplan_impl(int type, int dim, int *nmodes, int iflag, int ntran
 
     /* Automatically set GPU method. */
     if (d_plan->opts.gpu_method == 0) {
-        if (type == 1)
+        if (type == 1 && (sizeof(T) == 4 || dim < 3 || tol >= 1e-3))
```

Comment:
Can you add a comment explaining why this extra comparison is here. It will definitely elicit a WTF from a future reader

Reply:
There was an attempt on the next line. Will elaborate a bit more.